### PR TITLE
[IN-698] Setup cors

### PR DESCRIPTION
**Pull request by Fliko**
This gives a top level option `cors: true` it is false by default. The option will setup cors for every endpoint according to this material. https://enable-cors.org/server_awsapigateway.html

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -114,6 +114,46 @@ const getPathId = async ({ apig, apiId, endpoint }) => {
   return endpointFound ? endpointFound.id : null
 }
 
+const enableCORS = ({ endpoints }) => {
+  const defaultResponse = [{
+    code: 200,
+    headers: {
```
Comment (RealTschoegl) [inline utils.js:120]: Should these be made overwrite-able?

File: utils.js [hunk 2]
```diff
@@ -176,27 +216,46 @@ const mergeEndpointObjects = ({ endpoints, configEndpoints, stateEndpoints }) =>
...
-const isModified = ({ obj, previousObj }) => {
-  return !Object.keys(obj).every((prop) => {
-    // function or authorizer could be either an arn or function-name so check both
-    if ((prop === 'authorizer' || prop === 'function') && previousObj[prop]) {
-      return obj[prop] === previousObj[prop] || obj[prop] === previousObj[prop].split(':')[6]
+const compareProps = ({ obj, previousObj }) => {
+  // Both arrays and objects must have the same number of keys and each key be equal
+  if (Array.isArray(obj) && previousObj) {
+    return obj.length === previousObj.length
+      && obj.every((key, i) => compareProps({ obj: key, previousObj: previousObj[i] }))
```
Comment (RealTschoegl) [inline utils.js:223]: Fancy - .every()!

File: utils.js [hunk 3]
```diff
@@ -449,51 +516,44 @@ const createMethods = async ({ apig, apiId, endpoints }) => {
...
+      .catch(e => {
+        if (e.code === 'ConflictException') {
+          const params = {
+            httpMethod: endpoint.method,
+            resourceId: endpoint.id,
+            restApiId: apiId,
+            statusCode: `${response.code}`,
+            patchOperations:[{
+              op: 'replace',
+              path: '/responseParameters'
```
Comment (RealTschoegl) [inline utils.js:548]: Kinda curious what this does.

File: utils.js (hunk 1 above)
Comment (Fliko) [reply utils.js:120]: they are, those are just defaults, you can include them in the yml file to overwrite

File: utils.js (hunk 3 above)
Comment (Fliko) [reply utils.js:548]: Yeah so aws has an update for all their create functions. Put is more like a create or update. So update you use json pointer to identify a field on a resource that you want to modify. Didn't make sense to me until I found this https://docs.aws.amazon.com/apigateway/api-reference/link-relation/model-update/

**Pull request merged**